tests/quantum_sequencer_lite_tests.cpp

```diff
@@ -0,0 +1,525 @@
+/*
+** Copyright 2018 Bloomberg Finance L.P.
+**
+** Licensed under the Apache License, Version 2.0 (the "License");
+** you may not use this file except in compliance with the License.
+** You may obtain a copy of the License at
+**
+**     http://www.apache.org/licenses/LICENSE-2.0
+**
+** Unless required by applicable law or agreed to in writing, software
+** distributed under the License is distributed on an "AS IS" BASIS,
+** WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+** See the License for the specific language governing permissions and
+** limitations under the License.
+*/
+
+#include <quantum_fixture.h>
+#include <gtest/gtest.h>
+
+#ifdef BLOOMBERG_QUANTUM_SEQUENCER_LITE_SUPPORT
+
+using namespace quantum;
+
+//==============================================================================
+// TEST FIXTURES
+//==============================================================================
+
+struct SequencerLiteTest: public DispatcherFixture
+{};
+
+INSTANTIATE_TEST_CASE_P(SequencerLiteTest_Default,
+                        SequencerLiteTest,
+                        ::testing::Values(TestConfiguration(false, false),
+                                          TestConfiguration(false, true)));
+
+// Utility with common functionality for the sequencer related tests
+class SequencerLiteTestData
+{
+public: // types
+    using SequenceKey = int;
+    using TaskId = int;
+    using TaskSequencerLite = SequencerLite<SequenceKey>;
+    using TaskSequencerLiteConfiguration = SequencerLiteConfiguration<SequenceKey>;
+    struct TaskResult
+    {
+        /// Task start time
+        std::chrono::steady_clock::time_point startTime;
+        /// Task end time
+        std::chrono::steady_clock::time_point endTime;
+    };
+    using TaskResultMap = std::unordered_map<TaskId, TaskResult>;
+    using SequenceKeyMap = std::unordered_map<SequenceKey, std::vector<TaskId>>;
+
+    /**
+     * @Brief checks if a task was started before another task finished
+     */
+    void ensureOrder(TaskId beforeTaskId, TaskId afterTaskId)
+    {        
+        TaskResultMap::const_iterator beforeTaskIt = _results.find(beforeTaskId);
+        ASSERT_NE(beforeTaskIt, _results.end());
+        TaskResultMap::const_iterator afterTaskIt = _results.find(afterTaskId);
+        ASSERT_NE(afterTaskIt, _results.end());
+        if (beforeTaskIt == _results.end() || afterTaskIt == _results.end())
```

Comment:
The asserts above should make this if superfluous; might as well remove it.

Reply:
The asserts will likely not be present in production code.